sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/api/ApiExtensions.kt

```diff
@@ -309,3 +313,11 @@ fun Call.findCallInPrecedingCallChain(matcher: FunMatcher, bindingContext: Bindi
     }
     return receiver to receiverResolved
 }
+
+fun ResolvedValueArgument.isNull(bindingContext: BindingContext) = (
+    (this as? ExpressionValueArgument)
+        ?.valueArgument
+        ?.getArgumentExpression()
+        ?.predictRuntimeValueExpression(bindingContext)
+        as? KtConstantExpression
+    )?.isNull() ?: false
```

Comment:
`KtExpression` actually has `isNull()` defined on it (as extension method). So you should be able to remove the cast here.